r/R/dplyr-funcs-string.R

```diff
@@ -257,11 +269,13 @@ register_bindings_string_regex <- function() {
     if (opts$fixed) {
       out <- call_binding("startsWith", x = string, prefix = opts$pattern)
     } else {
-      out <- call_binding("grepl", pattern = paste0("^", opts$pattern), x = string, fixed = FALSE)
-    }
-
-    if (negate) {
-      out <- !out
+      out <- create_string_match_expr(
+        arrow_fun = "match_substring_regex",
+        string = string,
+        pattern = paste0("^", opts$pattern),
+        ignore_case = opts$ignore_case,
+        negate = negate
+      )
```

Comment:
Same thing here about negate as `str_ends` below (sorry the comments are backwards! I saw it there first 🙈 )